crates/core/src/operations/vacuum.rs

```diff
@@ -235,6 +245,43 @@ impl VacuumBuilder {
             None => Utc::now().timestamp_millis(),
         };
 
+        let keep_files = match &self.keep_versions {
+            Some(versions) => {
+                let max_version = self.snapshot.version();
```

Comment:
Rather than iterating over all versions from 0 to the max version and getting the files for every version, can you instead iterate over just the versions specified in `keep_versions`? (Maybe after sorting them first to simplify the state updates). Then you should be able to remove the `files_by_version` variable and just add files directly to `keep_files`.

Reply:
I can do that. The reason I wrote it this way was twofold:
1. Debugging the file dependency is tricky. By building an intermediate breakout, we can more clearly see the required files for each version. This helps diagnose future problems.
2. To get the file dependency for each version, the underlying code has to replay all log files up through max(keep_versions). So, the performance gain by switching the code is not all that great.

Reply:
That's fair, but I worry that for very large tables the hash map of files per version could get very large. On your comment about having to replay all log files, I think we can avoid some of that work by starting from the first version to keep rather than version 0, assuming snapshots can be used to avoid replaying all log files (I left an inline comment about this too).